src/MainWindow.vala

```diff
@@ -1177,9 +1161,13 @@ namespace Terminal {
             // Closing a tab will switch to another, which will trigger check for same names
         }
 
-        private void action_new_tab () {
+        public void action_new_tab () {
             if (Application.settings.get_boolean ("follow-last-tab")) {
-                new_tab (current_terminal.get_shell_location ());
+                if (current_terminal != null) {
+                    new_tab (current_terminal.get_shell_location ());
+                } else {
+                    new_tab (Environment.get_current_dir ());
+                }
```

Comment:
it's better to keep any "current directory" concept exclusive to the application class. because of the local/primary instance logic.

Reply:
So what do you suggest?

Reply:
Note: Environment.get_current_dir () is already in use in TerminalWidget

Reply:
> So what do you suggest?

Only the Application class known the real working directory for the current invocation, so either action-new-tab is updated to take a path, or we don't try  to use the action in the command line context, where is the only place that a concept of working directory exists.

> Note: Environment.get_current_dir () is already in use in TerminalWidget

It actually isn't, the only place that we call activate_shell() without a location is in the case that we call new_tab() with a empty location, what isn't possible.